lib/routes/pos/home/pos_invoice.dart

```diff
@@ -517,3 +578,46 @@ class POSInvoiceState extends State<POSInvoice> {
         ]).toList());
   }
 }
+
+class Amount {
```

Comment:
I am OK with have the parsing logic encapsulated in a separate class but I think this class should hold only:
1. The text entered by the user (as string variable)
2. The fiat conversion or the bitcoin currency.

It should provide calculated property for the text we show the user (after parsing).
This will simplify a lot of code at the widget level and put all the parsing logic here.